pint/models/frequency_dependent.py

```diff
@@ -0,0 +1,61 @@
+"""This module implements a frequency evolution of pulsar profiles model"""
+from warnings import warn
+import parameter as p
+from .timing_model import TimingModel, Cache
+import astropy.units as u
+import numpy as np
+import pint.utils as ut
+import astropy.time as time
+
+class FD(TimingModel):
+    """This class provides a timing model for frequency evolution of pulsar
+    profiles model.
+    """
+    def __init__(self):
+        super(FD, self).__init__()
+        self.add_param(p.prefixParameter(name='FD1', units="second", value=0.0,
+                       descriptionTplt=lambda x: ("%d term of frequency"
+                                                  " dependent  coefficients" % x),
+                       unitTplt=lambda x: 'second',
+                       type_match='float'))
+
+        self.delay_funcs['L1'] += [self.FD_delay]
+
+    def setup(self):
+        super(FD, self).setup()
+        # Check if FD terms are in order.
+        FD_terms = self.get_prefix_mapping('FD').keys()
+        FD_terms.sort()
+        FD_in_order = range(1,max(FD_terms)+1)
+        if not FD_terms == FD_in_order:
+            diff = list(set(FD_in_order) - set(FD_terms))
+            raise MissingParameter("Spindown", "FD%d"%diff[0])
+        self.num_FD_terms = len(FD_terms)
+
+    def FD_delay(self, toas):
+        """This is a function for calculation of frequency dependent delay.
+        See nanograv 9 years data set (2015) Eq.(2)
+        FDdelay = sum(c_i * (log(obs_freq/1GHz))^i)
+        """
+        FD_delay = np.zeros_like(toas, dtype=np.longdouble) * u.second
+        for ii in range(1,self.num_FD_terms+1):
+            FD_coef = getattr(self, self.FDmapping[ii])
+            log_freq = np.log(toas['freq'] / (1 * u.GHz))
+            FD_delay += FD_coef.value * (log_freq) ** ii
```

Comment:
It probably doesn't actually matter for the FD parameters, but it would be good if we were consistent in PINT about how we evaluate polynomials, since there are numerical precision issues you can run into.  In utils.py, there is `taylor_horner()` which is close to what you want, but not the same since it includes the factorials in the denominator appropriate for a Taylor expansion.   Maybe we should make a similar evaluate_polynomial() function where you specify the coefficients c_i and the factorials are not assumed?  This would allow those two functions to be optimized at some later date and all the places that use them would automatically get the benefit of the improved speed or accuracy. Thoughts? @scottransom @demorest @vhaasteren